tests/foreman/api/test_docker.py

```diff
@@ -1303,16 +1308,18 @@ def test_positive_create_with_name(self):
         """
         for name in valid_data_list():
             with self.subTest(name):
-                url = gen_url(subdomain=gen_string('alpha'))
                 description = gen_string('alphanumeric')
                 registry = entities.Registry(
                     description=description,
                     name=name,
-                    url=url,
+                    url=self.url,
                 ).create()
-                self.assertEqual(registry.name, name)
-                self.assertEqual(registry.url, url)
-                self.assertEqual(registry.description, description)
+                try:
+                    self.assertEqual(registry.name, name)
+                    self.assertEqual(registry.url, self.url)
+                    self.assertEqual(registry.description, description)
+                finally:
+                    registry.delete()
```

Comment:
1) Actually we already have separate class for registry tests =)
2) Of course, I thought about that approach first, but unfortunately, it cannot be applied to subtests, only for test entity level, so no chances to work in that way